src/witan/app/forecast.clj

```diff
@@ -459,16 +463,17 @@
   :allowed-methods #{:post}
   :processable? (fn [ctx]
                   (let [forecast (get-most-recent-version id)
-                        inputs (:inputs (util/get-post-params ctx))]
-                    (and forecast
-                         ((util/post!-processable-validation ws/UpdateForecast) ctx)
-                         (all-categories-exist-in-model? forecast (keys inputs))
-                         (every? data/exists? inputs)
-                         (or (-> forecast :public? not) ;; if public, check all data inputs are public
-                             (every? #(-> % second :public?) inputs)))))
+                        inputs   (:inputs (util/get-post-params ctx))
+                        tests    [[(fn [] forecast)                                                    "forecast is nil"]
```

Comment:
Agreed. Originally I had five tests but managed to condense it. Oh well, will remove. At least `reduced` worked well.